### PR TITLE
Ensure proper cleanup of watchfiles threads on server shutdown

### PR DESCRIPTION
**Pull request by philippjfr**
Makes sure that the `watchfiles` watcher threads are cleaned up by calling cleanup on all models before shutdown.

### CODE REVIEW

File: panel/io/server.py [hunk 1]
```diff
@@ -104,6 +104,8 @@
 INDEX_HTML = os.path.join(os.path.dirname(__file__), '..', '_templates', "index.html")
 DEFAULT_TITLE = "Panel Application"
 
+_running_tasks = set()
```
Comment (hoxbro) [inline panel/io/server.py:107]: Is this needed?

Comment (philippjfr) [reply panel/io/server.py:107]: No, not anymore.

File: panel/io/server.py [hunk 2]
```diff
@@ -104,6 +104,8 @@
...
+
```
Comment (philippjfr) [inline panel/io/server.py:108]:
```suggestion
```

File: panel/io/server.py [hunk 3]
```diff
@@ -1173,14 +1176,11 @@ def show_callback():
...
 
-    def sig_exit(*args, **kwargs):
-        server.io_loop.add_callback_from_signal(do_stop)
-
     def do_stop(*args, **kwargs):
         server.io_loop.stop()
 
     try:
-        signal.signal(signal.SIGINT, sig_exit)
+        server.io_loop.asyncio_loop.add_signal_handler(signal.SIGINT, do_stop)
```
Comment (philippjfr) [inline panel/io/server.py:1183]:
```suggestion
        signal.signal(signal.SIGINT, sig_exit)
```

**Pull request merged**